include/solvers/mpm_base.tcc

```diff
@@ -1477,11 +1477,30 @@ void mpm::MPMBase<Tdim>::particles_pore_pressures(
           reference_points.insert(std::make_pair<double, double>(
               static_cast<double>(position), static_cast<double>(h0)));
         }
+
+        // Read gravity
+        Eigen::Matrix<double, Tdim, 1> gravity =
+            Eigen::Matrix<double, Tdim, 1>::Zero();
+        auto loads = io_->json_object("external_loading_conditions");
+        if (loads.contains("gravity")) {
+          if (loads.at("gravity").is_array() &&
+              loads.at("gravity").size() == gravity.size()) {
+            for (unsigned i = 0; i < gravity.size(); ++i) {
+              gravity[i] = loads.at("gravity").at(i);
+            }
+          } else {
+            throw std::runtime_error("Specified gravity dimension is invalid");
+          }
+        } else {
+          throw std::runtime_error(
+              "In order to use the option water table, \"gravity\" should be "
+              "specified in the \"external_loading_conditions\"! ");
+        }
```

Comment:
Nice - I used the same snippet in the traction boundary BC for grabbing gravity!